### ISSUE TITLE
Default handling for empty response body?

### ISSUE THREAD

**Issue by airjp73** [rtk-incubator/rtk-query#89]
Hi all! I was trying out rtk-query by converting some old code and noticed something I thought might make sense to have some default support for.

In our app at work, all of our DELETE endpoints return a 204 with no response body. I noticed that, by default, empty responses from a request cause the mutation promise to reject. So this code always resulted in an error notification:

```js
const [deleteUser] = useDeleteUsersQuery();

const onDelete = () => {
  deleteUser(id)
    .then(unwrapResult)
    .then(/* success notification */)
    .catch(/* error notification */);
}
```

The cause is that the internal response handler unconditionally calls `res.json()` when the response comes back. 

This is possible to handle with a custom `responseHandler` argument in the api configuration:

```js
deleteUser: builder.mutation<void, string>({
  query: (id) => ({
    url: `/users/${id}`,
    method: 'DELETE',
    responseHandler: async () => null,
  }),
  invalidates: [USERS],
}),
```

So as-is, it is possible to handle this situation. But, I was wondering what you all thought about adding some default support for this case? I'm not sure how common of a use-case this is, but it does seem like it would be good to not reject by default.

If you like the idea, I'd love to take a crack at making the change myself.

**Comment by phryneas** [comment]
A shorter version that is currently is possible is `responseHandler: 'text'`.

But yeah, a PR that automatically handles the edge case of empty-string responses as empty string instead of trying to parse it as JSON would be welcome I guess - what do you think @msutkowski ?

**Comment by msutkowski** [comment]
Sounds reasonable to me!

**Comment by phryneas** [comment]
Just merged #91, thank you for bringing this to our attention & the PR :+1:
